### PR TITLE
Undo limit (100 history points)

### LINKED ISSUE #699: Undo history limit

### PR DESCRIPTION
**Pull request by mightyiam**
Fixes #699

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -541,7 +548,7 @@ module.exports = function (grunt) {
         grunt.task.run([
             'bower',
             'browserify:externalModules',
-            'browserify:keysim',
+            'browserify:modulesForTests',
```
Comment (mightyiam) [inline Gruntfile.js]: There was only a single and now there are two so I grouped them together.

File: package.json [hunk 2]
```diff
@@ -35,6 +35,7 @@
     }
   ],
   "devDependencies": {
+    "call-n-times": "^2.0.0",
```
Comment (mightyiam) [inline package.json]: Use da external things.

File: src/test/unit/specific_feature_tests/undo_redo.js [hunk 3]
```diff
@@ -358,3 +359,27 @@ test("Redo keyboard shortcut", function () {
...
+            // try going back 101 times
+            callNTimes(function () {
+                wymeditor.undoRedo.undo();
+            }, 101);
+        },
+        // if there was no limit or the limit was higher than 100, then
+        // it would have been `<p>99</p>`
+        expectedResultHtml: "<p>100</p>"
+    });
+});
```
Comment (mightyiam) [inline src/test/unit/specific_feature_tests/undo_redo.js]: I spent some time making this test readable enough. Is it?

File: Gruntfile.js [hunk 4]
```diff
@@ -210,6 +210,13 @@ module.exports = function (grunt) {
                     'edited.js'],
                 dest: '<%= yeoman.app %>/lib/edited.js'
             },
+            callNTimes: {
+                // this is not concatenated into the build.
+                // it is used only in tests.
+                src: ['<%= yeoman.app %>/wymeditor/external-modules/' +
```
Comment (winhamwr) [inline Gruntfile.js]: Do we want to switch to using lodash before merging this in?

File: src/test/unit/specific_feature_tests/undo_redo.js (hunk 3 above)
Comment (winhamwr) [inline src/test/unit/specific_feature_tests/undo_redo.js]: Looks readable to me. I like the counting to test the limit in a very explicit way.

**Comment by winhamwr** [comment]
This looks good, but maybe we should merge in once its switch to using the lodash helper?

File: package.json [hunk 5]
```diff
@@ -35,6 +35,7 @@
     }
   ],
   "devDependencies": {
+    "lodash.times": "^3.0.1",
```
Comment (mightyiam) [inline package.json]: Lodash has builds for individual things.

**Pull request merged**